File: mathematical_programs/python_programs/digit_counter.py
from sys import stdin, stdout
import math
def iterative_approach(number):
        counter = 0
        while number%10:
                number = int(number/10)
                counter += 1
        return counter

def recursive_approach(number):
        if not number:
                return 0
        return recursive_approach(int(number/10))+1

def logarithmic_approach(number):
        return math.floor(math.log(number, 10))+1

def to_string_approach(number):
        return len(str(number))

if __name__=="__main__":
        #number = stdin.readline()
        number = 48564561265456894569454616456426165465465465465
        stdout.write("iter: "+str(iterative_approach(number))+"\n")
        stdout.write("recur: "+str(recursive_approach(number))+"\n")
        stdout.write("log: "+str(logarithmic_approach(number))+"\n")
        stdout.write("to_str: "+str(to_string_approach(number))+"\n")

File: mathematicat_programs/c++_programs/digit_counter.cpp
#include <bits/stdc++.h>
using namespace std;
//#define loop(i, a, b) for (int i = a; i <= b; i++)
typedef long long int ll;

int iterative_approach(ll digit)
{
        int counter=0;
        while(digit%10)
        {
                counter++;
                digit /=10;
        }
        return counter;
}
int recursive_approach(ll digit)
{
        if (digit==0)
        {
                return 0;
        }
        return recursive_approach(digit/10)+1;
}
int logarithemic_approach(ll digit)
{
        return floor(log10(digit)+1);
}
int to_string_approach(ll digit)
{
        string number = to_string(digit);
        return number.size();
}
int main()
{
        ll number=144515545441454;
        printf("iter: %d\n", iterative_approach(number));
        printf("recur: %d\n", recursive_approach(number));
        printf("log: %d\n", logarithemic_approach(number));
        printf("to_str: %d", to_string_approach(number));
        return 0;
}

